### DISCUSSION TITLE
Why using 3 spaces as indentation?

### DISCUSSION THREAD

**Discussion by vvaltchev** [vvaltchev/tilck#88] [Q&A] [answered]
(Question moved from the main README.md file.)

**Comment by vvaltchev** [comment, marked as answer]
Long story. It all started after using that coding style for years at VMware. Initially, it looked pretty weird to me, but at some point I felt in love with the way code looked. I got convinced that 2 spaces are just not enough, while 4 spaces are somehow "too much". Therefore, when I started the project in 2016, I decided to stick with the indentation size I liked most, even if I knew that using 4 spaces would have been better for most people. Today, I'm not sure if that was the right decision, but I still like the way Tilck's code looks.

**Reply by 0xbadcaffe** [reply]
https://youtu.be/SsoOG6ZeyUI 😀

**Reply by vvaltchev** [reply]
Ahahahah, yeah I know that scene: it's hilarious!
Still, in my case even the "spaces" people "hate" me, because it's not 2, nor 4, nor 8 spaces.

**Reply by 0xbadcaffe** [reply]
Well as long as your spouse is not a programmer who's using number of spaces other than 3 I guess your fine :)

**Reply by themadsens** [reply]
FWIW. My previous employments (which lasted for 17 yrs) also specified 3 spaces indent in the style guide. Along with, horrors, all curly braces on separate lines etc.

My experience is the same as yours: It just looks better :-)

**Reply by vvaltchev** [reply]
@themadsens I'm happy to hear that I'm not alone :-)

**Comment by dandv** [comment]
Why isn't 3-space indentation a lot more popular? To quote from [this post](https://www.audero.it/blog/2015/10/21/the-revolution-of-3-spaces-code-indentation/),

> In my opinion, 2 spaces don’t give the code enough room to be readable. I feel that 2 spaces aren’t enough to understand, at a quick glance, where a nested block ends and when another starts. On the contrary, 4 spaces give a block too much “breath” and I feel this leads to the same issue I mentioned for the tab.

> 3 spaces seem just perfect. They give you enough space to quickly identify nested blocks, but not so much that after two or three level of nesting your code reaches the right margin of the screen.

People are [making fun of 3-space indent](https://www.reddit.com/r/ProgrammerHumor/comments/6mu36m/3_space_indentation/) as if it were patently bad and everyone should know why.

**Reply by vvaltchev** [reply]
Thanks for sharing, @dandv. Yeah, I believe that people don't like 3-space indentation, just because it's an odd number; as if there was a rule somewhere claiming that only even numbers could be used :-)
